lib/ansible/modules/cloud/vultr/vr_ssh_key_facts.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# (c) 2018, Yanis Guenane <yanis+ansible@guenane.org>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: vr_ssh_key_facts
+short_description: Gather facts about the Vultr SSH keys available.
+description:
+  - Gather facts about SSH keys available.
+version_added: "2.7"
+author: "Yanis Guenane (@Spredzy)"
+extends_documentation_fragment: vultr
+'''
+
+EXAMPLES = r'''
+- name: Gather Vultr SSH keys facts
+  local_action:
+    module: vr_ssh_key_facts
+
+- name: Print the gathered facts
+  debug:
+    var: ansible_facts.vultr_ssh_key_facts
+'''
+
+RETURN = r'''
+---
+vultr_api:
+  description: Response from Vultr API with a few additions/modification
+  returned: success
+  type: complex
+  contains:
+    api_account:
+      description: Account used in the ini file to select the key
+      returned: success
+      type: string
+      sample: default
+    api_timeout:
+      description: Timeout used for the API requests
+      returned: success
+      type: int
+      sample: 60
+    api_retries:
+      description: Amount of max retries for the API requests
+      returned: success
+      type: int
+      sample: 5
+    api_endpoint:
+      description: Endpoint used for the API requests
+      returned: success
+      type: string
+      sample: "https://api.vultr.com"
+ansible_facts:
+  description: Response from Vultr API
+  returned: success
+  type: complex
+  contains:
+    "vultr_ssh_key_facts": [
+      {
+        "SSHKEYID": "5abf426403479",
```

Comment:
the key is now `id`

~~~
        "id": "5abf426403479",
~~~

Reply:
Its actually both. "id" and "SSHKEYID", right?

Reply:
I see, I missed this in 577427b1c20dd27bdcd42e417d9ffccfb669b374 that the origin key is also going to be returned as we do not delete the key after "renaming" it. This is going to be done with all the current modules, so every module would have a new return

My intention was, to just rename the key to something "simple": *id*.
Let's delete the previous key in resource in module_utils/vultr.py once renamed, ok?

~~~patch
diff --git a/lib/ansible/module_utils/vultr.py b/lib/ansible/module_utils/vultr.py
index 0dd185c73c..f52052502c 100644
--- a/lib/ansible/module_utils/vultr.py
+++ b/lib/ansible/module_utils/vultr.py
@@ -238,6 +238,7 @@ class Vultr:
 
                 if 'key' in config:
                     resource[config['key']] = resource[search_key]
+                    del resource[search_key]
 
         return resource
~~~

Reply:
done